src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelCopyRatioEmissionData.java

```diff
@@ -0,0 +1,130 @@
+ *     {@link #poissonAdditiveNoise} = [read depth]_s x [mapping error rate]_{st}
+ * </p>
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelCopyRatioEmissionData implements Serializable {
+
+    private static final long serialVersionUID = -7363264674200250712L;
+
+    /**
+     * Mean log multiplicative bias
+     */
+    private final double mu;
+
+    /**
+     * Total unexplained variance
+     */
+    private final double psi;
+
+    /**
+     * Raw read count
+     */
+    private final int readCount;
+
+    /**
+     * Additive noise in the Poisson parameter
+     */
+    private final double poissonAdditiveNoise;
+
+    /**
+     * Sample metadata (sex genotype, depth of coverage, ...)
+     *
+     * @implNote The metadata is necessary for calculating the emission probability, though, they are the same
+     * for all targets. We have separated such global parameters from target-dependent variables and wrapped them
+     * in a class so that they can be "injected" by reference.
+     */
+    private CopyRatioCallingMetadata copyRatioCallingMetadata = null;
+
+    /**
+     * Public constructor.
+     *
+     * @param mu Mean log multiplicative bias
+     * @param psi Total unexplained variance
+     * @param readCount Raw read count
+     * @param poissonAdditiveNoise Additive noise in the Poisson parameter
+     */
+    public CoverageModelCopyRatioEmissionData(final double mu,
+                                              final double psi,
+                                              final int readCount,
+                                              final double poissonAdditiveNoise) {
+        this.mu = ParamUtils.isFinite(mu, "Log multiplicative bias must be finite. Bad value: " + mu);
+        this.psi = ParamUtils.isPositiveOrZero(psi, "Unexplained variance must be a non-negative real number." +
+                " Bad value: " + psi);
+        this.readCount = ParamUtils.isPositiveOrZero(readCount, "Read count must be a non-negative real number." +
+                " Bad value: " + readCount);
+        this.poissonAdditiveNoise = ParamUtils.isPositiveOrZero(poissonAdditiveNoise, "Additive noise in the Poisson" +
+                " parameter must be non-negative. Bad value: " + poissonAdditiveNoise);
+    }
+
+    public double getMu() {
+        return mu;
+    }
+
+    public double getPsi() {
+        return psi;
+    }
+
+    public int getReadCount() {
+        return readCount;
+    }
+
+    public double getPoissonAdditiveNoise() {
+        return poissonAdditiveNoise;
+    }
+
+    public CopyRatioCallingMetadata getCopyRatioCallingMetadata() {
+        return Utils.nonNull(copyRatioCallingMetadata, "Sample metadata is queried but is not set");
+    }
+
+    /**
+     * Set metadata.
+     *
+     * @param copyRatioCallingMetadata the metadata
+     */
+    public void setCopyRatioCallingMetadata(@Nonnull final CopyRatioCallingMetadata copyRatioCallingMetadata) {
```

Comment:
It's a hack to avoid serializing of the same object multiple times. Neither the java serializer, nor Kryo, is smart enough to serialize just a single instance of `CopyRatioCallingMetadata` even if all entries in `List<CoverageModelCopyRatioEmissionData>` have the reference to the same instance of `CopyRatioCallingMetadata`. So, I don't initialize the `CopyRatioCallingMetadata` field in the beginning, serialize `List<CoverageModelCopyRatioEmissionData>`, and inject the metadata later; that's why `CopyRatioExpectationsCalculator<STATE, DATA>.getCopyRatioPosteriorExpectations(CopyRatioCallingMetadata, List<Target>, List<DATA>)` has such a signature.